R/process_sc_all_care_home.R

```diff
@@ -150,62 +177,39 @@ process_sc_all_care_home <- function(
     ) %>%
     dplyr::summarise(
       sc_latest_submission = dplyr::first(.data[["period"]]),
-      dplyr::across(
-        c(
-          "ch_discharge_date",
-          "ch_provider",
-          "record_date",
-          "qtr_start",
-          "ch_name",
-          "ch_postcode",
-          "reason_for_admission",
-          "type_of_admission"
-        ),
-        dplyr::first
-      ),
+      dplyr::across(c(
+        "ch_discharge_date",
+        "ch_provider",
+        "period_end_date",
+        "period_start_date",
+        "ch_name",
+        "ch_postcode",
+        "reason_for_admission",
+        "type_of_admission"
+      ), dplyr::first),
       dplyr::across(c("gender", "dob", "postcode"), dplyr::first)
     ) %>%
-    dplyr::ungroup() %>%
-    # Amend dates for split episodes
-    # Change the start and end date as appropriate when an episode is split,
-    # using the start / end date of the submission quarter
-    dplyr::group_by(
-      .data[["chi"]],
-      .data[["sending_location"]],
-      .data[["social_care_id"]],
-      .data[["ch_admission_date"]]
-    ) %>%
-    # counter for latest submission
-    # TODO check if this is the same as split_episode_counter?
-    dplyr::mutate(
-      latest_submission_counter = tidyr::replace_na(
-        .data[["sc_latest_submission"]] != dplyr::lag(
-          .data[["sc_latest_submission"]]
-        ),
-        TRUE
-      ),
-      sum_latest_submission = cumsum(.data[["latest_submission_counter"]])
-    ) %>%
+    # If the admission date is missing use the period start date
+    # otherwise use the start of the quarter
     dplyr::mutate(
-      # If it's the first episode(s) then keep the admission date(s),
-      # otherwise use the start of the quarter
-      ch_admission_date = dplyr::if_else(
-        .data[["sum_latest_submission"]] == min(.data[["sum_latest_submission"]]),
-        .data[["ch_admission_date"]],
-        .data[["qtr_start"]]
+      ch_admission_date = dplyr::if_else(is.na(.data[["ch_admission_date"]]),
+        .data[["period_start_date"]],
+        .data[["ch_admission_date"]]
       ),
       # If it's the last episode(s) then keep the discharge date(s), otherwise
       # use the end of the quarter
-      ch_discharge_date = dplyr::if_else(
-        .data[["sum_latest_submission"]] == max(.data[["sum_latest_submission"]]),
-        .data[["ch_discharge_date"]],
-        .data[["record_date"]]
+      ch_discharge_date = dplyr::if_else(is.na(.data[["ch_discharge_date"]]),
+        .data[["period_end_date"]],
+        .data[["ch_discharge_date"]]
       )
     ) %>%
-    dplyr::ungroup()
+    dplyr::ungroup() %>%
+    dplyr::select(-period_start_date, -split_episode_counter)
+
 
   # Compare to Deaths Data
   # match ch_episode data with deaths data
+  # TO DO should this be boxi nrs death dates instead of IT extract deaths?
```

Comment:
Yes - i do think this should be using the BOXI NRS dates instead of NRS dates given in the IT extract. However we will need to wait to make this change once we have further info on the deaths and a team decision is made

Reply:
agree. made new issue for it